src/Compilers/CSharp/Portable/FlowAnalysis/ExitPointsWalker.cs

```diff
@@ -142,6 +142,7 @@ override protected void LeaveRegion()
                     case BoundKind.YieldReturnStatement:
                     case BoundKind.AwaitExpression:
                     case BoundKind.UsingStatement:
+                    case BoundKind.ForEachStatement when ((BoundForEachStatement)pending.Branch).AwaitOpt != null:
```

Comment:
>case [](start = 20, length = 4)

:memo: This was causing a crash. Covered by `TestControlFlowAnalysis`.

Reply:
maybe update the comment explaining why?

---
In reply to: [199896338](https://github.com/dotnet/roslyn/pull/28218#discussion_r199896338) [](ancestors = 199896338)